### PR TITLE
Expand description and results of the test pipeline.

### PR DESCRIPTION
**Pull request by adelton**
## Description

Expand description and results of the test pipeline.

## How Has This Been Tested?

Looked at https://github.com/adelton/opendatahub-ai-edge/tree/test-pipeline/pipelines.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by adelton** [comment]
Conceptually this needs https://github.com/opendatahub-io/ai-edge/pull/117.

**Comment by adelton** [comment]
If the purpose of the `test-mlflow-image` pipeline is to test the behaviour of the container image and push the image to its "final" location, it is not clear why the applications and services stay around even after `skopeo-copy` finished.

Some clarification of the purpose of the `metrics` Service might also be useful.

File: pipelines/README.md [hunk 1]
```diff
@@ -102,13 +102,41 @@ oc secret link pipeline rhoai-edge-build-pull-secret
...
+and/or [`bike-rentals-test-data-cm.yaml`](tekton/test-mlflow-image-pipeline/bike-rentals-test-data-cm.yaml):
 
 ```bash
 oc apply -k tekton/test-mlflow-image-pipeline/
 oc create -f tekton/test-mlflow-image-pipeline/test-mlflow-image-pipelinerun-tensorflow-housing.yaml
+# and/or
+oc create -f tekton/test-mlflow-image-pipeline/test-mlflow-image-pipelinerun-bike-rental.yaml
+```
+
+Check what pipeline was created and run and what applications started either in OpenShift Console in Pipelines > Pipelines, Topology > select the application and its Resources and Pod logs, or with CLI find the names of the objects with
```
Comment (piotrpdev) [inline pipelines/README.md:122]:
```suggestion
Check what pipeline was created and what applications were started either using:
* OpenShift Console in Pipelines > Pipelines, Topology > select the application and its Resources and Pod logs.

  OR

* CLI; find the names of the objects with:
```

File: pipelines/README.md [hunk 2]
```diff
@@ -102,13 +102,41 @@ oc secret link pipeline rhoai-edge-build-pull-secret
...
 
-> **NOTE**
-> Make sure to change the `target-imagerepo` parameter to match the name of your Quay namespace if using one of the provided `PipelineRun` files.
+Update the `target-imagerepo` parameter value from its default `rhoai-edge` in
+[`test-mlflow-image-pipelinerun-tensorflow-housing.yaml`](tekton/test-mlflow-image-pipeline/test-mlflow-image-pipelinerun-tensorflow-housing.yaml)
+and/or
+[`test-mlflow-image-pipelinerun-bike-rental.yaml`](tekton/test-mlflow-image-pipeline/test-mlflow-image-pipelinerun-bike-rental.yaml)
+to match the name of your Quay namespace.
+
+Then create the pipeline(s) to test the container images behaviour with
```
Comment (piotrpdev) [inline pipelines/README.md:111]:
```suggestion
Then create the pipeline(s) to test the container images behaviour using
```

**Review by piotrpdev — APPROVED** [review]

File: pipelines/README.md (hunk 1 above)
Comment (adelton) [reply pipelines/README.md:122]: This seems to suggest one cannot use both methods.

File: pipelines/README.md (hunk 2 above)
Comment (adelton) [reply pipelines/README.md]: Thanks, updated in 69c0aa71394c8642c8fb6a1ce0f278ae019605b7.

**Comment by adelton** [comment]
@piotrpdev Thanks for the approval.

Should I file a separate issue for the question about the applications and services above, to have the reasoning properly discussed?

File: pipelines/README.md (hunk 1 above)
Comment (piotrpdev) [reply pipelines/README.md:140]: I said OR not XOR 😄 but sure I updated it.

**Comment by piotrpdev** [comment]
> @piotrpdev Thanks for the approval.
> 
> Should I file a separate issue for the question about the applications and services above, to have the reasoning properly discussed?

Sure if you want, here's the current reasoning:

<br>

> If the purpose of the `test-mlflow-image` pipeline is to test the behaviour of the container image and push the image to its "final" location, it is not clear why the applications and services stay around even after `skopeo-copy` finished.

This is what Stefan's pipeline code did (which the current pipeline code is based on) and we just haven't updated it to clean up after itself yet (and it's been used a lot for dev testing), feel free to make an issue for this.

<br>

> Some clarification of the purpose of the `metrics` Service might also be useful.

It was added back in #48 to expose metrics for the deployed image but we have a [template for that now](https://github.com/opendatahub-io/ai-edge/blob/62bb5f6a9c38f7b08a8e85eed2996addf876e3d2/acm/odh-edge/base/model-service.yaml). Pretty sure it's obsolete and can be removed unless we decide to test the metrics port using `test-mlflow-image-pipeline`. I doubt it though since we're planning to [completely change how testing is done](https://github.com/opendatahub-io/ai-edge/pull/55#issuecomment-1706997731).

**Comment by adelton** [comment]
Thanks.

I've filed https://github.com/opendatahub-io/ai-edge/pull/127 for the cleanup and https://github.com/opendatahub-io/ai-edge/pull/128 to avoid creating the metrics service.

**Pull request merged**